tesseract_time_parameterization/src/time_optimal_trajectory_generation.cpp

```diff
@@ -1093,137 +1090,105 @@ bool Trajectory::isValid() const { return valid_; }
 
 double Trajectory::getDuration() const { return trajectory_.back().time_; }
 
-bool Trajectory::assignData(TrajectoryContainer& trajectory, double path_tolerance) const
+bool Trajectory::assignData(TrajectoryContainer& trajectory) const
 {
-  auto it = trajectory_.begin();
+  Eigen::VectorXd path_lengths = Eigen::VectorXd::Zero(trajectory.size());
+  double length = 0;
+  for (Eigen::Index i = 1; i < trajectory.size(); ++i)
+  {
+    length += (trajectory.getPosition(i) - trajectory.getPosition(i - 1)).norm();
+    path_lengths[i] = length;
+  }
+
+  if (tesseract_common::almostEqualRelativeAndAbs(length, 0))
+    throw std::runtime_error("Tried to assign trajectory with length zero.");
+
+  double scale = path_.getLength() / length;
+  path_lengths = scale * path_lengths;
```

Comment:
Updated.

``` cpp
  // The length of the joint interpolated trajectory is not the same length
  // of the TOTG path because it using cubic spline so this scales the joint
  // interpolated path length matches the TOTG length.
```